interface/ceed-preconditioning.c

```diff
@@ -688,9 +698,11 @@ static int CeedSingleOperatorAssemble(CeedOperator op, CeedInt offset, CeedVecto
 
   // Loop over elements and put in data structure
   // We store B_mat_in, B_mat_out, BTD, elem_mat in row-major order
-  CeedSize    count = 0;
-  CeedScalar *vals, *BTD_mat = NULL, *elem_mat = NULL, *elem_mat_b = NULL;
+  CeedSize           count = 0;
+  CeedTensorContract contract;
+  CeedScalar        *vals, *BTD_mat = NULL, *elem_mat = NULL, *elem_mat_b = NULL;
 
+  CeedCall(CeedBasisGetTensorContract(basis_in, &contract));
```

Comment:
This is not guaranteed to work in general. We need some logic to handle when the basis doesn't have a TensorContract object. Anything in the interface level files must work for any backend.

Reply:
OK, is it sufficient then to move `CeedMatrixMatrixMultiply` back to a non-static function and just call that implementation when `contract = NULL`?

Reply:
Yeah, that would probably be the simplest fix?

Reply:
Fixed, see below.